src/main/java/de/hysky/skyblocker/skyblock/chocolatefactory/ChocolateFactorySolver.java

```diff
@@ -209,10 +216,20 @@ private static Optional<Rabbit> getCoach(ItemStack coachItem) {
 		}
 
 		Matcher costMatcher = COST_PATTERN.matcher(coachLore);
+		Matcher levelMatcher = COACH_LEVEL_PATTERN.matcher(coachLore);
 		OptionalLong cost = RegexUtils.getLongFromMatcher(costMatcher, multiplierIncreaseMatcher.hasMatch() ? multiplierIncreaseMatcher.end() : 0); //Cost comes after the multiplier line
+		int level;
+		try {
+			level = RomanNumerals.romanToDecimal(String.valueOf(levelMatcher.find()));
+		} catch (IllegalArgumentException e) {
+			level = 0;
+		}
```

Comment:
`Matcher#find` returns `true` or `false` based on whether the regex matched. See https://docs.oracle.com/javase/8/docs/api/java/util/regex/Matcher.html#find--.
```suggestion
		int level = 0;
		if (levelMatcher.find()) {
			level = RomanNumerals.romanToDecimal(levelMatcher.group(1));
		}
```